driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -611,6 +615,21 @@ void cleanupIdleConnections(long now) {
         }
     }
 
+    @Override
+    protected void checkNotInEventLoop() {
+        if (!CHECK_IO_DEADLOCKS)
+            return;
+        for (EventExecutor executor : cluster.manager.connectionFactory.eventLoopGroup) {
+            if (executor.inEventLoop()) {
+                throw new IllegalStateException(
```

Comment:
Oh yes, I explained that, but on the jira ticket:

> After looking further into this, I think just warning makes no sense. Synchronous calls on an I/O thread are always a bug, and I can't think of any situation where we would want to leave users' applications running with a risk of deadlock, with potentially very hard to debug consequences.
So I'm going for the exception directly.

Funny that your suggestion is exactly how I had done it before changing my mind :)